src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -0,0 +1,1322 @@
+
+        VectorPart vp = (VectorPart) p;
+          
+        //iterate over command list
+        for (VectorCommand cmd : vp.getCommandList())
+        {
+          //There are three types of commands: MOVETO, LINETO and SETPROPERTY
+          switch (cmd.getType())
+          {
+            case LINETO:
+            case MOVETO:
+            {
+              /**
+               * Move the laserhead (laser on) from the current position to the x/y position of this command.
+               */
+              if (first_vector) {
+                first_vector = false;
+
+                stream.hex("ca0100");
+                stream.hex("ca02").byteint(part_number); // start_layer
+                stream.hex("ca0113"); // blow on
+                stream.hex("c902").longint((int)speed);
+                // power for laser #1
+                stream.hex("c601").percent((int)min_power);
+                stream.hex("c602").percent((int)max_power);
+
+                /* start vector mode */
+                stream.hex("ca030f");
+                stream.hex("ca1000");
+              }
+              vector(cmd.getX(), cmd.getY(), p.getDPI(), cmd.getType() == CmdType.LINETO);
+              break;
+            }
+            case SETPROPERTY:
+            {
+              LaserProperty pr = cmd.getProperty();
+              if (pr instanceof FloatMinMaxPowerSpeedFrequencyProperty)
+              {
+                FloatMinMaxPowerSpeedFrequencyProperty prop = (FloatMinMaxPowerSpeedFrequencyProperty) pr;
+                if (first_prop) {
+                  first_prop = false;
+                  min_power = cmd_layer_percent("c631", part_number, min_power, prop.getMinPower());
+                  max_power = cmd_layer_percent("c632", part_number, max_power, prop.getPower());
+                  speed = cmd_layer_absoluteMM("c904", part_number, speed, prop.getSpeed());
+                  // focus - n/a
+                  // frequency
+                  stream.hex("c660").byteint(part_number).hex("00").longint(prop.getFrequency());
+                  // color - red for now
+                  long color = (0 << 16) + (0 << 8) + 100;; //(normalizeColor(this.blue) << 16) + (normalizeColor(this.green) << 8) + normalizeColor(this.red);
+                  stream.hex("ca06").byteint(part_number).longint(color);
+                  // CA 41
+                  stream.hex("ca41").byteint(part_number).byteint(0);
+                }
+                else {
+                  min_power = cmd_percent("c601", min_power, prop.getMinPower());
+                  max_power = cmd_percent("c602", max_power, prop.getPower());
+                  speed = cmd_absoluteMM("c902", speed, prop.getSpeed());
+                }
+              }
+              break;
+            }
+            default:
+            {
+              System.out.println("*** Ruida unknown vector part(" + cmd.getType() + ")");
```

Comment:
here we have println, later we have warnings.add for something similar.

Reply:
fixed